hadoop-hdds/docs/content/concept/OzoneS3Gateway.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Ozone S3 Gateway"
+date: "2025-03-28"
+weight: 8
+menu: 
+  main:
+     parent: Architecture
+summary: Apache Ozone’s S3 Gateway (often called S3G) is a component that provides an Amazon S3-compatible REST interface for Ozone’s object store.
+---
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+Apache Ozone’s S3 Gateway (often called **S3G**) is a component that provides an Amazon S3-compatible REST interface for Ozone’s object store. In essence, it allows any S3 client or tool to read and write data in Ozone as if it were talking to AWS S3. The S3 Gateway runs as a **separate, stateless service** on top of Ozone, translating HTTP S3 API calls into Ozone’s native storage operations.
+
+This design lets users leverage the rich ecosystem of S3-compatible applications with Ozone, without modifying those applications.
+
+## High-Level Overview of the S3 Gateway
+
+Ozone’s S3 Gateway acts as a **REST front-end** that speaks the S3 protocol and bridges it to the Ozone backend. It is **stateless**, meaning it does not persist data or metadata locally between requests – all state is stored in the Ozone cluster itself.
+
+Being stateless allows multiple S3 Gateway instances to run in parallel behind a load balancer for scaling and high availability. The gateway supports a broad set of S3 operations – create/delete buckets, PUT/GET objects, list objects, perform multipart uploads – using standard S3 SDKs or CLI tools.
+
+Under the hood, each operation is translated into the appropriate Ozone calls:
+- A PUT object request becomes a `putKey` write operation.
+- A GET object becomes a `getKey` read operation.
+
+### Mapping S3 to Ozone Concepts
+
+In Ozone, data is organized into **volumes**, **buckets**, and **keys**. By default, the S3 Gateway uses a special volume named `/s3v` to store all S3 buckets.
```

Comment:
This is not entirely accurate as the volume name can be changed. Also, bucket linking needs to be discussed.